charts/osm/templates/validatingwebhook.yaml

```diff
@@ -0,0 +1,29 @@
+apiVersion: admissionregistration.k8s.io/v1beta1
+kind: ValidatingWebhookConfiguration
+metadata:
+  labels:
+    app: osm-controller
+  name: {{.Values.OpenServiceMesh.webhookConfigNamePrefix}}-{{.Values.OpenServiceMesh.meshName}}
+webhooks:
+- name: validating-webhook.k8s.io
+  clientConfig:
+    service:
+      name: osm-controller
+      namespace: {{.Release.Namespace}}
+      path: /validate-webhook
+      port: 443
+  failurePolicy: Fail
+  matchPolicy: Exact
+  namespaceSelector:
+    matchLabels:
+      name: osm-system
+  rules:
+    - apiGroups:
+        - ""
+      apiVersions:
+        - v1
+      operations:
+        - CREATE
+        - UPDATE
```

Comment:
We should validate DELETE too to ensure values that should always exist in the ConfigMap are never deleted.

Reply:
@shashankram, for DELETE, in the osm configmap, what fields, do you think, would the user be ok in deleting?

Reply:
I haven't looked into this much, but I think whatever the charts create during install should be preserved, ex the following should probably exist in the ConfigMap at all times:

```
  permissive_traffic_policy_mode: {{ .Values.OpenServiceMesh.enablePermissiveTrafficPolicy | default "false" | quote }}
  egress: {{ .Values.OpenServiceMesh.enableEgress | quote }}
  envoy_log_level: {{ .Values.OpenServiceMesh.envoyLogLevel | quote }}
  enable_debug_server: {{ .Values.OpenServiceMesh.enableDebugServer | quote }}
  prometheus_scraping: {{ .Values.OpenServiceMesh.enablePrometheusScraping | quote }}
  use_https_ingress: {{ .Values.OpenServiceMesh.useHTTPSIngress | default "false" | quote }}
  service_cert_validity_duration: {{ .Values.OpenServiceMesh.serviceCertValidityDuration | quote }}
```

If you want to handle DELETE scenarios later, that's totally acceptable. Lets create an issue for this and tackle it later?